// frontend/src/club-testing.js
export const clubData = [
    {
      id: 1,
      location: "Ashgrove",
      address: "123 Ashgrove Road, Ashgrove QLD 4060",
      phone: "(07) 3355 6677",
      email: "ashgrove@hsg.com.au",
      latitude: "-27.445528",
      longitude: "152.992028",
      facilities: [
        "State-of-the-art cardio and strength equipment",
        "Group fitness classes (Yoga, Pilates, HIIT, Boxing, Zumba)",
        "Personal training",
        "Boxing and karate studios",
        "24-hour access",
        "On-site café"
      ]
    },
    {
      id: 2,
      location: "Brisbane City",
      address: "456 Queen Street, Brisbane City QLD 4000",
      phone: "(07) 3366 7788",
      email: "brisbanecity@hsg.com.au",
      latitude: "-27.471028",
      longitude: "153.025333",
      facilities: [
        "Spacious gym floor with a wide range of equipment",
        "Group fitness classes (Yoga, Pilates, HIIT, Boxing, Zumba)",
        "Personal training",
        "Boxing and karate studios",
        "24-hour access",
        "On-site café"
      ]
    },
    {
      id: 3,
      location: "Chermside",
      address: "789 Chermside Road, Chermside QLD 4032",
      phone: "(07) 3377 8899",
      email: "chermside@hsg.com.au",
      latitude: "-27.385917",
      longitude: "153.028694",
      facilities: [
        "Outdoor training area with functional fitness equipment",
        "Group fitness classes (Yoga, Pilates, HIIT, Boxing, Zumba)",
        "Personal training",
        "Boxing and karate studios",
        "24-hour access",
        "On-site café"
      ]
    },
    {
      id: 4,
      location: "Graceville",
      address: "1011 Graceville Road, Graceville QLD 4075",
      phone: "(07) 3388 9900",
      email: "graceville@hsg.com.au",
      latitude: "-27.522806",
      longitude: "152.980333",
      facilities: [
        "Indoor pool with lap lanes and spa",
        "Group fitness classes (Yoga, Pilates, HIIT, Boxing, Zumba)",
        "Personal training",
        "Boxing and karate studios",
        "24-hour access",
        "On-site café"
      ]
    },
    {
      id: 5,
      location: "Westlake",
      address: "1213 Westlake Drive, Westlake QLD 4074",
      phone: "(07) 3399 0011",
      email: "westlake@hsg.com.au",
      latitude: "-27.545028",
      longitude: "152.918111",
      facilities: [
        "Rooftop terrace with outdoor workout area",
        "Group fitness classes (Yoga, Pilates, HIIT, Boxing, Zumba)",
        "Personal training",
        "Boxing and karate studios",
        "24-hour access",
        "On-site café"
      ]
    }
  ];
