// frontend/src/data-testing.js
export const classData = [
  // Monday (02-10-2024) - This Week
  {
    id: 1,
    class: "PILATES",
    instructor: ["Jannie Home","jack Home"],
    date: "02-10-2024",
    time: "07:00", // 24-hour format
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 10,
  },
  {
    id: 2,
    class: "ZUMBA",
    instructor: ["Jane Doe"],
    date: "02-10-2024",
    time: "13:00", // 24-hour format
    duration: 60,
    location: "Brisbane",
    capacity: 25,
    booked: 15,
  },
  {
    id: 3,
    class: "BOXING",
    instructor: ["John Smith"],
    date: "02-10-2024",
    time: "18:00", // 24-hour format
    duration: 75,
    location: "Brisbane",
    capacity: 20,
    booked: 8,
  },

  // Tuesday (03-10-2024) - This Week
  {
    id: 4,
    class: "INDOOR CYCLING",
    instructor: ["Jane Home", "John Doe"],
    date: "03-10-2024",
    time: "07:00",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 10,
  },
  {
    id: 5,
    class: "HIIT",
    instructor: ["Emily Brown"],
    date: "03-10-2024",
    time: "12:00",
    duration: 45,
    location: "Brisbane",
    capacity: 25,
    booked: 20,
  },
  {
    id: 6,
    class: "YOGA",
    instructor: ["Sophia Davis"],
    date: "03-10-2024",
    time: "19:30",
    duration: 60,
    location: "Brisbane",
    capacity: 35,
    booked: 12,
  },

  // Wednesday (04-10-2024) - This Week
  {
    id: 7,
    class: "ABS",
    instructor: ["Jack Home"],
    date: "04-10-2024",
    time: "09:00",
    duration: 30,
    location: "Brisbane",
    capacity: 30,
    booked: 18,
  },
  {
    id: 8,
    class: "SPIN",
    instructor: ["Olivia Wilson"],
    date: "04-10-2024",
    time: "14:00",
    duration: 45,
    location: "Brisbane",
    capacity: 28,
    booked: 22,
  },
  {
    id: 9,
    class: "PILATES",
    instructor: ["Jannie Home"],
    date: "04-10-2024",
    time: "17:30",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 16,
  },

  // Thursday (05-10-2024) - This Week
  {
    id: 10,
    class: "ZUMBA",
    instructor: ["Jane Doe"],
    date: "05-10-2024",
    time: "06:30",
    duration: 45,
    location: "Brisbane",
    capacity: 35,
    booked: 25,
  },
  {
    id: 11,
    class: "BOXING",
    instructor: ["John Smith"],
    date: "05-10-2024",
    time: "13:00",
    duration: 60,
    location: "Brisbane",
    capacity: 20,
    booked: 12,
  },
  {
    id: 12,
    class: "HIIT",
    instructor: ["Emily Brown"],
    date: "05-10-2024",
    time: "18:00",
    duration: 45,
    location: "Brisbane",
    capacity: 25,
    booked: 18,
  },

  // Friday (06-10-2024) - This Week
  {
    id: 13,
    class: "YOGA",
    instructor: ["Sophia Davis"],
    date: "06-10-2024",
    time: "07:00",
    duration: 60,
    location: "Brisbane",
    capacity: 40,
    booked: 28,
  },
  {
    id: 14,
    class: "ABS",
    instructor: ["Jack Home"],
    date: "06-10-2024",
    time: "12:30",
    duration: 30,
    location: "Brisbane",
    capacity: 30,
    booked: 24,
  },
  {
    id: 15,
    class: "SPIN",
    instructor: ["Olivia Wilson"],
    date: "06-10-2024",
    time: "17:45",
    duration: 45,
    location: "Brisbane",
    capacity: 28,
    booked: 16,
  },

  // Saturday (07-10-2024) - Next Week
  {
    id: 16,
    class: "BOXING",
    instructor: ["Jack Home"],
    date: "07-10-2024",
    time: "09:30",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 10,
  },
  {
    id: 17,
    class: "PILATES",
    instructor: ["Jannie Home"],
    date: "07-10-2024",
    time: "13:00",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 22,
  },
  {
    id: 18,
    class: "ZUMBA",
    instructor: ["Jane Doe"],
    date: "07-10-2024",
    time: "18:30",
    duration: 60,
    location: "Brisbane",
    capacity: 35,
    booked: 28,
  },

  // Sunday (08-10-2024) - Next Week
  {
    id: 19,
    class: "YOGA",
    instructor: ["Jack Home"],
    date: "08-10-2024",
    time: "10:00",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 10,
  },
  {
    id: 20,
    class: "HIIT",
    instructor: ["Emily Brown"],
    date: "08-10-2024",
    time: "14:30",
    duration: 45,
    location: "Brisbane",
    capacity: 25,
    booked: 16,
  },
  {
    id: 21,
    class: "SPIN",
    instructor: ["Olivia Wilson"],
    date: "08-10-2024",
    time: "17:00",
    duration: 45,
    location: "Brisbane",
    capacity: 28,
    booked: 20,
  },
  // additional classes for the next week
  {
    id: 22,
    class: "YOGA",
    instructor: ["Sophia Davis"],
    date: "09-10-2024",
    time: "07:00",
    duration: 60,
    location: "Brisbane",
    capacity: 40,
    booked: 12,
  },
  {
    id: 23,
    class: "ABS",
    instructor: ["Jack Home"],
    date: "01-10-2024",
    time: "12:30",
    duration: 30,
    location: "Brisbane",
    capacity: 30,
    booked: 5,
  },
  {
    id: 24,
     class: "SPIN",
    instructor: ["Olivia Wilson"],
    date: "01-10-2024",
    time: "17:45",
    duration: 45,
    location: "Brisbane",
    capacity: 28,
    booked: 18,
  },
  {
    id: 25,
    class: "BOXING",
    instructor: ["Jack Home"],
    date: "30-09-2024",
    time: "09:30",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 8,
  },
  {
    id: 26,
    class: "PILATES",
    instructor: ["Jannie Home"],
    date: "30-09-2024",
    time: "13:00",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 15,
  },
  {
    id: 27,
    class: "YOGA",
    instructor: ["Jannie Home"],
    date: "02-10-2024",
    time: "07:00",
    duration: 45,
    location: "Brisbane",
    capacity: 30,
    booked: 15,
  },
];
